STAC sortby extension does not always sort

I just got this failure in the CI on a PR that only touched lines after the tests. All other tests, both before and after `test_stac_sortby_extension[default]`, passed.
```python
_____________________ test_stac_sortby_extension[default] ______________________

stac_client = <FlaskClient <Flask 'cubedash'>>

    @pytest.mark.parametrize("env_name", ("default",), indirect=True)
    def test_stac_sortby_extension(stac_client: FlaskClient) -> None:
        sortby = [{"field": "properties.datetime", "direction": "asc"}]
        rv: Response = stac_client.post(
            "/stac/search",
            data=json.dumps(
                {
                    "product": "ga_ls8c_ard_3",
                    "time": "2022-01-01T00:00:00/2022-12-31T00:00:00",
                    "limit": OUR_DATASET_LIMIT,
                    "_full": True,
                    "sortby": sortby,
                }
            ),
            headers={"Content-Type": "application/json", "Accept": "application/json"},
        )
        assert rv.status_code == 200
        doc = rv.json
        assert doc.get("features")
        for i in range(1, len(doc["features"])):
            assert (
                doc["features"][i - 1]["properties"]["datetime"]
                < doc["features"][i]["properties"]["datetime"]
            )
    
        sortby = [{"field": "properties.datetime", "direction": "desc"}]
        rv: Response = stac_client.post(
            "/stac/search",
            data=json.dumps(
                {
                    "product": "ga_ls8c_ard_3",
                    "time": "2022-01-01T00:00:00/2022-12-31T00:00:00",
                    "limit": OUR_DATASET_LIMIT,
                    "_full": True,
                    "sortby": sortby,
                }
            ),
            headers={"Content-Type": "application/json", "Accept": "application/json"},
        )
        assert rv.status_code == 200
        doc = rv.json
        for i in range(1, len(doc["features"])):
            assert (
                doc["features"][i - 1]["properties"]["datetime"]
                > doc["features"][i]["properties"]["datetime"]
            )
    
        rv: Response = stac_client.get(
            "/stac/search?collection=ga_ls8c_ard_3&limit=5&sortby=assets"
        )
        assert rv.status_code == 400
    
        rv: Response = stac_client.get(
            "/stac/search?collection=ga_ls8c_ard_3&limit=5&sortby=id,-properties.datetime"
        )
        doc = rv.json
        for i in range(1, len(doc["features"])):
            assert doc["features"][i - 1]["id"] < doc["features"][i]["id"]
    
        # use of property prefixes shouldn't impact result
        rv: Response = stac_client.get(
            "/stac/search?collection=ga_ls8c_ard_3&limit=5&sortby=item.id,-datetime"
        )
        assert rv.json == doc
    
        # ignore undefined field
        rv: Response = stac_client.get(
            "/stac/search?collection=ga_ls8c_ard_3&limit=5&sortby=id,-datetime,foo"
        )
        assert rv.json["features"] == doc["features"]
    
        # sorting across pages
        next_link = _get_next_href(doc)
        next_link = next_link.replace("http://localhost", "")
        rv: Response = stac_client.get(next_link)
        last_item = doc["features"][-1]
        next_item = rv.json["features"][0]
>       assert last_item["id"] < next_item["id"]
E       AssertionError: assert '2a7c420c-dc07-42bf-ae91-ad5da19c76f6' < '0f294559-b2cf-423b-98a8-57a5ed1b2dbd'
```

--- a/cubedash/_stac.py
+++ b/cubedash/_stac.py
@@ -513,7 +513,12 @@ def _sort_arg(arg: str | list[str] | list[dict[str, Any]]) -> list[dict[str, Any
         # default is ascending
         return {"field": val.strip(), "direction": "asc"}
 
-    arg_work = arg.split(",") if isinstance(arg, str) else arg
+    if isinstance(arg, str):
+        arg_work = (
+            json.loads(arg.replace("'", '"')) if arg.startswith("[") else arg.split(",")
+        )
+    else:
+        arg_work = arg
     if len(arg_work):
         if isinstance(arg_work[0], str):
             return [_format(a) for a in arg_work]  # type: ignore[arg-type]
@@ -643,8 +648,8 @@ def _handle_search_request(
             _o=next_offset,
             _full=full_information,
             intersects=intersects,
-            fields=fields,
-            sortby=sortby,
+            fields=str(fields) if fields else None,
+            sortby=str(sortby) if sortby else None,
             # so that it doesn't get named 'filter_lang'
             **{"filter-lang": filter_lang},
             filter=filter_cql,

--- a/cubedash/_stac.py
+++ b/cubedash/_stac.py
@@ -648,7 +648,7 @@ def _handle_search_request(
             _o=next_offset,
             _full=full_information,
             intersects=intersects,
-            fields=str(fields) if fields else None,
+            fields=fields,
             sortby=str(sortby) if sortby else None,
             # so that it doesn't get named 'filter_lang'
             **{"filter-lang": filter_lang},

--- a/cubedash/_stac.py
+++ b/cubedash/_stac.py
@@ -521,7 +521,7 @@ def _sort_arg(arg: str | list[str] | list[dict[str, Any]]) -> list[dict[str, Any
         arg_work = arg
     if len(arg_work):
         if isinstance(arg_work[0], str):
-            return [_format(a) for a in arg_work]  # type: ignore[arg-type]
+            return [_format(a) for a in arg_work]
         if isinstance(arg_work[0], dict):
             for a in arg_work:
                 assert isinstance(a, dict)

